core/test/run_models_test.jl

```diff
@@ -476,3 +476,28 @@ end
         0,
     ) ≈ 5.0 atol = 0.001 skip = Sys.isapple()
 end
+
+@testitem "mean_flow" begin
+    using DataFrames: DataFrame
+
+    toml_path =
+        normpath(@__DIR__, "../../generated_testmodels/flow_boundary_time/ribasim.toml")
+    @test ispath(toml_path)
+    function get_flow(solver_adaptive::Bool, solver_saveat::Float64)::Vector{Float64}
+        config = Ribasim.Config(toml_path)
+        solver_dt = solver_adaptive ? nothing : config.solver.dt
+        config = Ribasim.Config(toml_path; solver_dt, solver_saveat)
+        model = Ribasim.run(config)
+        df = DataFrame(Ribasim.flow_table(model))
+        return filter(
+            [:from_node_id, :to_node_id] => (from, to) -> from === 3 && to === 2,
+            df,
+        ).flow_rate
+    end
+    @test all(get_flow(true, 86400.0) .≈ 1.0)
+    @test all(get_flow(true, 0.0) .≈ 1.0)
+    @test all(get_flow(true, Inf) .≈ 1.0)
```

Comment:
Would be good to also test the length of `tsaves` here, to make sure they are in fact different. And also test `saveat` values that don't fit n times in the tspan, to trigger the remainder path.